web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSet.jsx

```diff
@@ -560,12 +562,17 @@ export class ResultSetUtils {
   }
 
   processClipboardVal(columnVal, col, rawCopiedVal) {
-    if(columnVal === '') {
+    // if value is blank or the column is primary key
+    if(columnVal === '' || col.isPK) {
+      // if column has default value
       if(col.has_default_val) {
         columnVal = undefined;
-      } else if(rawCopiedVal === null) {
+      } else if(rawCopiedVal === null || col.isPK) {
         columnVal = null;
       }
+    } else if (col.has_default_val && col.seqtypid) {
+      // if column has default value and is serieal type
+      columnVal = undefined;
```

Comment:
It seems to me that this should behave as pgAdmin III did, and ask the user what they want to do. What if I specifically want to paste the original value(s)? That is perfectly valid thing to do in some circumstances.

Reply:
In which situation, the user wants to paste the original value(s)? It may be a very rare situation and for that do we need to ask every time while pasting whenever we have a serial type?

Reply:
Copying data from one table to another, maybe from a test system to prod or vice versa. Or when testing something, to re-populate some deleted data by copying original values back from. a spreadsheet or similar.

I wouldn't object to having a "Don't ask me again" option or similar.

Reply:
Or we can have the option "Copy Serials" the same as "Copy with headers".

Reply:
> Or we can have the option "Copy Serials" the same as "Copy with headers".

with default as false, I agree. Better than popup.

Reply:
@dpage I have added an option in paste, attached screenshot

<img width="1219" alt="Screenshot 2023-04-24 at 2 06 50 PM" src="https://user-images.githubusercontent.com/111979571/233944369-f331f2e9-f5e1-4209-8d7c-6485679101d3.png">

By default this option will be unchecked and if a user selects this option and then paste serial values will be pasted. Any suggestion/thoughts ?

Reply:
Is "serials" the best wording here? Maybe "Paste with SERIAL/IDENTITY values?"

Reply:
Updated screen 
<img width="1219" alt="Screenshot 2023-04-24 at 2 52 45 PM" src="https://user-images.githubusercontent.com/111979571/234213383-227bbc50-aa41-45f7-9e59-81f18061dac9.png">